addons/l10n_jo_edi_extended/models/account_move.py

```diff
@@ -0,0 +1,80 @@
+from odoo import _, api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_jo_edi_invoice_type = fields.Selection(
+        selection=[
+            ('local', 'Local'),
+            ('export', 'Export'),
+            ('development', 'Development Area'),
+        ],
+        string="Invoice Type",
+        help="Invoice Types as per the Income and Sales Tax Department for JoFotara",
+        precompute=True,
+        compute='_compute_l10n_jo_edi_invoice_type',
+        readonly=False,
+        store=True,
+        tracking=True,
+    )
+    l10n_jo_edi_state = fields.Selection(selection_add=[('demo', 'Sent (Demo)')])
+
+    @api.depends('partner_id.country_code')
+    def _compute_l10n_jo_edi_invoice_type(self):
+        for move in self.filtered(lambda m: m.l10n_jo_edi_is_needed and m.l10n_jo_edi_invoice_type != 'development'):
+            country_code = move.partner_id.country_code
+            if country_code == 'JO':
+                move.l10n_jo_edi_invoice_type = 'local'
+            elif country_code:
+                move.l10n_jo_edi_invoice_type = 'export'
+            else:
+                move.l10n_jo_edi_invoice_type = False
+
+    @api.depends('partner_id', 'company_id')
+    def _compute_preferred_payment_method_line_id(self):
+        super()._compute_preferred_payment_method_line_id()
+
+        for move in self.filtered(lambda m: m.partner_id and m.l10n_jo_edi_is_needed):
+            candidate_journals = self.env['account.journal'].search([
+                ('type', 'in', ('bank', 'cash')),
+                ('company_id', '=', move.company_id.id),
+            ])
+            expected_type = 'bank' if move.partner_id.is_company or move.partner_id.parent_id else 'cash'
```

Comment:
In most localizations we just do a separate selection field for this.  (where it asks a payment method on the invoice, but in most cases, nothing has been paid yet)  What bothers me a lot is that we create new payment method lines on purpose that besides reporting here do not seem to have a special function.